tensorboard/webapp/app_routing/route_config.ts

```diff
@@ -166,6 +220,16 @@ class RouteConfigMatcher {
       };
     }
 
+    if (this.config.routeType === RouteType.PROGRAMMATIC_REDIRECTION) {
```

Comment:
optional: now that we have 3 types of route configs, L210 might read better as `if (routeDefType === REDIRECTION && this.redirectionFragments)`, in case readers don't realize that `redirectionFragments` are specific to only the REDIRECTION route def type.

Reply:
Definitely refactored and improved.